### ISSUE TITLE
Does MyBatis work inside Freemarker?

### ISSUE THREAD

**Issue by slady** [mybatis/freemarker-scripting#33]
I use MyBatis-Freemarker plugin as described at: http://www.mybatis.org/freemarker-scripting/
I configured MyBatis to work with Freemarker like this:
        <setting name="defaultScriptingLanguage" value="FreeMarker"/>
    </settings>
    <typeAliases>
        <typeAlias type="org.mybatis.scripting.freemarker.FreeMarkerLanguageDriver" alias="FreeMarker"/>
    </typeAliases>

The Freemarker scripts work just fine. The problem is I am unable to mix MyBatis tags inside a Freemarker code. I have some selects in MyBatis that I need to convert to Freemarker:

    <select id="find" parameterType="map" resultType="com.example.MyType">
        SELECT
            <if test="_databaseId == 'sqlserver'">
                TOP (${r"#{rowLimit}"})
            </if>
        FROM myexampletable
        <choose>
            <when test="_databaseId == 'oracle'">
                WHERE rownum &lt;= ${r"#{rowLimit}"}
            </when>
            <when test="_databaseId == 'postgresql' || _databaseId == 'h2'">
                LIMIT ${r"#{rowLimit}"}
            </when>
        </choose>
    </select>

(The above code simplified for the sake of brevity.)

The problem is this does not work at all, the MyBatis tags are ignored completely and all the TOP and LIMIT keywords appear in the same SQL command, no matter what DB dialect I use.

I have to say it had worked perfectly fine, before I added Freemarker scripting. Is this a bug or a feature? Can I make both languages work in one script at the same time?

Thank you for your kind answer!

**Comment by harawata** [comment]
No, you cannot use both XML and Freemaker syntax in the same statement.

**Comment by harawata** [comment]
Oops, submitted the form accidentally.

I've never used Freemaker myself, so please see the test cases.
It seems that you need to use `<#if  />` instead of `<if />`, for example.
https://github.com/mybatis/freemarker-scripting/blob/master/src/test/resources/sql/preparedIn.ftl

**Comment by slady** [comment]
No worries, I got the message. Your answer was short and concise. :)

The trouble is I wanted to mix both Freemaker syntax with MyBatis XML statements. I understand I can use Freemaker `<#if />` statements, this works perfectly well. The reason is I wanted to port the existing MyBatis code to Freemarker statements and I needed to make use of `_databaseId` variable that is set in MyBatis only. I know I can set variables to Freemarker as well, but that would be an extra hassle. And more would appear in the future.

I guess this feature is not going to change in the nearest future... Ooo, a bummer! I revert back to pure MyBatis.

Thanks for your reply! ;)
